components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -107,63 +172,145 @@ private static InternalAPI createAPI(String classFQName) {
+        return internalHttpsApiList;
+    }
+
+    public static boolean isSslConfiguredSuccessfully() {
+        return sslConfiguredSuccessfully;
+    }
 
-    public static int getInternalInboundPort() {
+    /**
+     * Reads and check from the synapse properties file whether the Internal api is enabled.
+     *
+     * @return - whether internal api is enabled in synapse properties file.
+     */
+    public static boolean isInternalApiEnabled() {
 
         File synapseProperties = Paths.get(CarbonUtils.getCarbonConfigDirPath(), "synapse.properties").toFile();
         Properties properties = new Properties();
-        InputStream inputStream = null;
-        try {
-            inputStream = new FileInputStream(synapseProperties);
+        try (InputStream inputStream = new FileInputStream(synapseProperties)) {
             properties.load(inputStream);
         } catch (FileNotFoundException e) {
             handleException("synapse.properties file not found", e);
         } catch (IOException e) {
             handleException("Error while reading synapse.properties file", e);
-        } finally {
-            if (inputStream != null) {
-                try {
-                    inputStream.close();
-                } catch (IOException ignored) {
-                }
-            }
         }
-        int internalInboundPort = Constants.DEFAULT_INTERNAL_HTTP_API_PORT;
-
         String internalInboundEnabledProperty = properties.getProperty(Constants.INTERNAL_HTTP_API_ENABLED);
         if (internalInboundEnabledProperty == null) {
-            return -1;
+            return false;
         }
-        boolean internalInboundEnabled = Boolean.parseBoolean(internalInboundEnabledProperty);
-        if (!internalInboundEnabled) {
-            return -1;
+        boolean isEnabled = Boolean.parseBoolean(internalInboundEnabledProperty);
+        if (isEnabled) {
+            internalInboundHttpPortProperty = properties.getProperty(Constants.INTERNAL_HTTP_API_PORT);
+            internalInboundHttpsPortProperty = properties.getProperty(Constants.INTERNAL_HTTPS_API_PORT);
         }
+        return isEnabled;
+    }
 
-        String internalInboundPortProperty =
-                properties.getProperty(Constants.INTERNAL_HTTP_API_PORT);
-        if (internalInboundPortProperty != null) {
-            try {
-                internalInboundPort = Integer.parseInt(internalInboundPortProperty);
-            } catch (Exception ex) {
-                handleException(Constants.INTERNAL_HTTP_API_PORT + " is not in proper format", ex);
+    private static SSLConfiguration setSslConfig(OMElement sslConfig) {
+
+        Iterator iterator = sslConfig.getChildElements();
+
+        String trustStore = null;
+        String keyStore = null;
+        String clientAuth = null;
+        String httpsProtocols = null;
+        String revocationVerifier = null;
+        String sslProtocol = null;
+        String prefferedCiphers = null;
```

Comment:
I suppose we need to initialized these values with what comes in the default configuration rather than null

Reply:
Since we have a default config coming with the pack we don't need to initalzie , if we do and if some one edits it wrongly also , then we would have to start with default configs. If they are editing they need to provide correct config and it is okay to fail startup IMO.

Reply:
If a provided param value is incorrect the server startup should fail, but if a parameter is not provided we need to work with the defaults, is what I feel. These parameters are all optional aren't they?

Reply:
No keystore is mandatory   . Others can be optional so NPe